app/src/main/java/com/sunasterisk/dmealfoodapp/ui/detailmeal/MealDetailActivity.kt

```diff
@@ -4,19 +4,153 @@ import android.content.Context
 import android.content.Intent
 import android.os.Bundle
 import com.google.android.youtube.player.YouTubeBaseActivity
+import com.google.android.youtube.player.YouTubeInitializationResult
+import com.google.android.youtube.player.YouTubePlayer
+import com.sunasterisk.dmealfoodapp.R
 import com.sunasterisk.dmealfoodapp.data.model.Meal
+import com.sunasterisk.dmealfoodapp.data.model.MealDetail
+import com.sunasterisk.dmealfoodapp.data.repository.MealRepository
+import com.sunasterisk.dmealfoodapp.data.source.local.MealLocalDataSource
+import com.sunasterisk.dmealfoodapp.data.source.local.dao.MealDaoImpl
+import com.sunasterisk.dmealfoodapp.data.source.local.db.AppDatabase
+import com.sunasterisk.dmealfoodapp.data.source.remote.MealRemoteDataSource
 import com.sunasterisk.dmealfoodapp.databinding.ActivityMealDetailBinding
+import com.sunasterisk.dmealfoodapp.utils.*
 
-class MealDetailActivity : YouTubeBaseActivity() {
+class MealDetailActivity : YouTubeBaseActivity(), MealDetailContact.View {
 
     private lateinit var binding: ActivityMealDetailBinding
+    private var presenter: MealDetailPresenter? = null
+    private var meal: Meal? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMealDetailBinding.inflate(layoutInflater)
         setContentView(binding.root)
+        initData()
+        initActions()
     }
 
+    override fun showMealDetailByMeal(mealDetails: List<MealDetail>) {
+        mealDetails.first {
+            binding.apply {
+                textNameMeal.text = it.name
+                textDescription.text = it.instruction
+                textIngredientSpecific.text = it.ingredient
+                showVideo(getIdVideo(it.youtube))
+            }
+            true
+        }
+    }
+
+    override fun showMealFavorite(isFavorite: Int) {
+        when {
+            isFavorite >= 1 -> {
+                binding.apply {
+                    buttonFavorite.show()
+                    buttonUnFavorite.gone()
+                }
+            }
+            else -> binding.apply {
+                buttonUnFavorite.show()
+                buttonFavorite.gone()
+            }
+        }
```

Comment:
1. Nếu có 2 case trở xuống, dùng if else, 3 case trở lên dùng when 
```suggestion
    val favorite = isFavorite >= 1
    binding.buttonFavorite.isVisible = favorite
    binding.buttonUnFavorite.isVisible = !favorite
```